src/equations/serre_green_naghdi_1d.jl

```diff
@@ -28,26 +59,36 @@ References for the Serre-Green-Naghdi system can be found in
 
 The semidiscretization implemented here conserves
 - the total water mass (integral of h) as a linear invariant
-- the total momentum (integral of h v) as a nonlinear invariant
-- the total energy
+- the total momentum (integral of h v) as a nonlinear invariant if the bathymetry is constant
+- the total modified energy
 
-for periodic boundary conditions, see
+for periodic boundary conditions (see Ranocha and Ricchiuto (2024)).
+Additionally, it is well-balanced for the lake-at-rest stationary solution, see
 - Hendrik Ranocha and Mario Ricchiuto (2024)
   Structure-preserving approximations of the Serre-Green-Naghdi
   equations in standard and hyperbolic form
   [arXiv: 2408.02665](https://arxiv.org/abs/2408.02665)
 """
 struct SerreGreenNaghdiEquations1D{Bathymetry <: AbstractBathymetry, RealT <: Real} <:
        AbstractSerreGreenNaghdiEquations{1, 3}
-    bathymetry::Bathymetry # type of bathymetry
+    bathymetry_type::Bathymetry # type of bathymetry
     gravity::RealT # gravitational constant
     eta0::RealT # constant still-water surface
 end
 
-function SerreGreenNaghdiEquations1D(; gravity_constant, eta0 = 0.0)
-    eta0 == 0.0 ||
-        @warn "The still-water surface needs to be 0 for the Serre-Green-Naghdi equations"
-    SerreGreenNaghdiEquations1D(bathymetry_flat, gravity_constant, eta0)
+function SerreGreenNaghdiEquations1D(bathymetry_type = bathymetry_flat;
```

Comment:
Honest question: Do you think it's best to make `bathymetry_flat` the default rather than `bathymetry_variable`? I know that it would be breaking to change this, but would it make sense to have the most general case the default?

Reply:
Let's see whether we need other breaking changes when introducing the hyperbolic approximation? I'm fine making the most general (but most expensive) case the default

Reply:
Ok, we can keep it like this for now, but maybe change this in the future when we have a breaking change anyway. We will have a breaking change anyway once we add the `bathymetry_type` also to the other equations and thus merge `BBMBBMEquations1D` and `BBMBBMVariableEquations1D` in one equation. I'm planning to do this once we are finished with integrating SGN.

Reply:
I have tracked this in https://github.com/JoshuaLampert/DispersiveShallowWater.jl/issues/130#issuecomment-2294812733. So I'm resolving this conversation.